tools/measurement/latency/src/types.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as moment from 'moment';
+
+import { TEST_TARGET_ID_MAP } from './test_target_id_map';
+
+export interface ProwJob {
+  apiVersion: string;
+  kind: string;
+  metadata: {
+    name: string;
+  };
+  spec: {
+    job: string;
+    refs: {
+      pulls?: Array<{ number: number }>;
+    };
+    type: string;
+  };
+  status: {
+    startTime: string;
+    completionTime: string;
+    state: string;
+  };
+}
+
+export class ReplaceMeasurement extends Array<string | number> {
```

Comment:
I'm not sure this should extend Array. It's probably best as a proper class (with fields) or as a function that simply returns the array. Can you add a method to ProwJob that converts it into the measurement array? That's another option.

Reply:
I'm not sure I agree. Isn't `ReplaceMeasurement` essentially an array?

JavaScript object and the its JSON serialization are almost the same syntactically. For example, `{'a': 123, 'b': 'hello world!'}` is both a valid JSON object and a valid JavaScript object literal. I think it's more burden to process mentally if we break this association, and replace it with our customized way of JSON serialization for an object.

Reply:
After all "JSON" is short for "JavaScript Object Notation".